ironfish/src/primitives/transaction.ts

```diff
@@ -79,7 +79,11 @@ export class Transaction {
       // proof
       reader.seek(PROOF_LENGTH)
 
-      return new NoteEncrypted(reader.readBytes(ENCRYPTED_NOTE_LENGTH, true))
+      const note = new NoteEncrypted(reader.readBytes(ENCRYPTED_NOTE_LENGTH, true))
+      // TODO(joe): remove once rpk removed from proof and put on transaction
```

Comment:
why the note has rpk? the sender address size should be updated in `ENCRYPTED_NOTE_LENGTH` right?

Reply:
In this section we are actually deserializing the `OutputDescription` which are called "notes" in this context 😅

Reply:
https://vscode.dev/github/iron-fish/ironfish/blob/07f483eec85cf88bd017849e4a2c5fdb729c4668/ironfish-rust-nodejs/src/structs/note_encrypted.rs#L30
pub const ENCRYPTED_NOTE_LENGTH: u32 =
    NOTE_ENCRYPTION_KEY_LENGTH + ENCRYPTED_NOTE_PLAINTEXT_LENGTH + 96;
   
pub const ENCRYPTED_NOTE_PLAINTEXT_LENGTH: u32 = ENCRYPTED_NOTE_SIZE as u32 + MAC_LENGTH;

pub const ENCRYPTED_NOTE_SIZE: usize =
    SCALAR_SIZE + MEMO_SIZE + AMOUNT_VALUE_SIZE + GENERATOR_SIZE + PUBLIC_ADDRESS_SIZE;
https://github.com/iron-fish/ironfish/blob/07f483eec85cf88bd017849e4a2c5fdb729c4668/ironfish-rust/src/note.rs#L27-L28

The encrypted note size is already includes public address size which is the sender address

Reply:
ugh I see, can you rename it so its less confusing.

Reply:
I am not sure we can. I think outputs are generally referred to as notes everywhere in typescript.